backend-shared/config/db/guardrow_test.go

```diff
@@ -0,0 +1,890 @@
+			Expect(applicationfirst.Name).ShouldNot(Equal(applicationsecond.Name))
+
+			rowsAffected, err := dbq.DeleteApplicationById(ctx, applicationsecond.Application_id)
+			Expect(err).To(BeNil())
+			Expect(rowsAffected).Should(Equal(1))
+
+			err = dbq.GetApplicationById(ctx, &applicationfirst)
+			Expect(err).To(BeNil())
+
+			err = dbq.GetApplicationById(ctx, &applicationsecond)
+			Expect(true).To(Equal(db.IsResultNotFoundError(err)))
+
+		})
+
+		It("Should test guard row against update and delete for applicationstates", func() {
+			err := db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			ctx := context.Background()
+			dbq, err := db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+			defer dbq.CloseDatabase()
+
+			_, managedEnvironment, _, gitopsEngineInstance, _, err := db.CreateSampleData(dbq)
+			Expect(err).To(BeNil())
+
+			applicationfirst := db.Application{
+				Application_id:          "test-my-application-1",
+				Name:                    "my-application",
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			err = dbq.CreateApplication(ctx, &applicationfirst)
+			Expect(err).To(BeNil())
+
+			applicationStatefirst := &db.ApplicationState{
+				Applicationstate_application_id: applicationfirst.Application_id,
+				Health:                          "Progressing",
+				Sync_Status:                     "Unknown",
+				Resources:                       make([]byte, 10),
+			}
+
+			err = dbq.CreateApplicationState(ctx, applicationStatefirst)
+			Expect(err).To(BeNil())
+
+			applicationsecond := db.Application{
+				Application_id:          "test-my-application-2",
+				Name:                    "my-application",
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+			err = dbq.CreateApplication(ctx, &applicationsecond)
+			Expect(err).To(BeNil())
+
+			applicationStatesecond := &db.ApplicationState{
+				Applicationstate_application_id: applicationsecond.Application_id,
+				Health:                          "Progressing",
+				Sync_Status:                     "Unknown",
+				Resources:                       make([]byte, 10),
+			}
+
+			err = dbq.CreateApplicationState(ctx, applicationStatesecond)
+			Expect(err).To(BeNil())
+			applicationStatesecond = &db.ApplicationState{
+				Applicationstate_application_id: applicationsecond.Application_id,
+				Health:                          "Progressing",
+				Sync_Status:                     "Sync",
+				Resources:                       make([]byte, 10),
+			}
+
+			err = dbq.UpdateApplicationState(ctx, applicationStatesecond)
+			Expect(err).To(BeNil())
+			Expect(applicationStatesecond.Sync_Status).Should(Equal("Sync"))
```

Comment:
You will need to do a dbq.GetApplicationState() here, before doing the comparison, to make sure the `applicationStatesecond` object has been updated with the database. (The go object doesn't get updated automatically)